spec/support/shared/functional/windows_script.rb

```diff
@@ -52,6 +54,8 @@
   shared_examples_for "a script resource with architecture attribute" do
     context "with the given architecture attribute value" do
       let(:resource_architecture) { architecture }
+      # TODO: Why does expected_architecture default to :i386 and not
+      # the machine architecture?
```

Comment:
I don't understand this default. The WindowsScript provider (which is the parent class of PowershellScript) is initialized with

```ruby
      def initialize( new_resource, run_context, script_extension='')
        super( new_resource, run_context )
        @script_extension = script_extension

        target_architecture = new_resource.architecture.nil? ?
          node_windows_architecture(run_context.node) : new_resource.architecture

        @is_wow64 = wow64_architecture_override_required?(run_context.node, target_architecture)

        @is_forced_32bit = forced_32bit_override_required?(run_context.node, target_architecture)
      end
```

and `node_windows_architecture` is given as

```ruby
      def node_windows_architecture(node)
        node[:kernel][:machine].to_sym
      end
```

It seems like the default should be the machine architecture and not `:i386`.